odoo/addons/test_new_api/tests/test_onchange.py

```diff
@@ -1345,3 +1349,16 @@ def test_new_one2many_traversing_many2one_second_onchange(self):
                 # check that Msg1 is visible in the one2many during onchange()
                 self.assertEqual(message_form.has_important_sibling, True)
                 message_form.body = 'Msg3: New'
+
+
+@tagged('post_install', '-at_install')
+class TestComputeWithoutDependencies(TransactionCase):
+    def test_compute_methods(self):
+        for model in self.env.values():
+            if model._abstract or not model._auto:
+                continue
+            for field in model._fields.values():
+                if field.compute and not self.env.registry.field_depends[field]:
+                    with self.subTest(msg=f"Compute method of {field} should work on new records"):
+                        with self.env.cr.savepoint():
+                            model.new()[field.name]
```

Comment:
Maybe you can use Form view to avoid tons of false positives error (although if you will still get a lot of them I suppose) ?

Reply:
What false positives are you talking about?

Reply:
I mean that some fields only works with its particular Form view. But obviously it means that is badly designed and compute should work on New record.
Anyway the number of compute that crashes is quite scary.

Reply:
Some of the problematic computed fields are simply not in a form view.  Those are therefore never actually used with new records.